rotkehlchen/db/upgrades/v36_v37.py

```diff
@@ -31,25 +39,64 @@ def _update_history_events_schema(write_cursor: 'DBCursor', conn: 'DBConnection'
     notes TEXT,
     type TEXT NOT NULL,
     subtype TEXT NOT NULL,
-    counterparty TEXT,
-    extra_data TEXT,
     FOREIGN KEY(asset) REFERENCES assets(identifier) ON UPDATE CASCADE,
     UNIQUE(event_identifier, sequence_index)
     );""")
     new_entries = []
+    dropped_data = []  # Store the dropped data to insert it into evm_events_info table later.
     with conn.read_ctx() as read_cursor:
         read_cursor.execute('SELECT * from history_events')
 
         for entry in read_cursor:
             if entry[11] is None:
-                new_entries.append(entry[:11] + ('none',) + entry[12:])
+                new_entries.append(entry[:11] + ('none',))
             else:
-                new_entries.append(entry)
+                new_entries.append(entry[:12])
+
+            if entry[12] is not None or entry[13] is not None:
+                dropped_data.append((entry[0], entry[12], entry[13]))
 
-    write_cursor.executemany('INSERT INTO history_events_copy VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ? ,? ,? ,?)', new_entries)  # noqa: E501
+    write_cursor.executemany('INSERT INTO history_events_copy VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)', new_entries)  # noqa: E501
     write_cursor.execute('DROP TABLE history_events')
     write_cursor.execute('ALTER TABLE history_events_copy RENAME TO history_events')
     log.debug('Exit _update_history_events_schema')
+    return dropped_data
+
+
+def _create_evm_events_info(write_cursor: 'DBCursor', dropped_data: DROPPED_DATA_TYPE) -> None:
+    """
+    Creates evm_events_info table and populates moves there values previously stored
+    in history_events.
+    """
+    log.debug('Enter _create_evm_events_info')
+    write_cursor.execute("""
+        CREATE TABLE IF NOT EXISTS evm_events_info(
+            identifier INTEGER PRIMARY KEY,
+            protocol TEXT,
+            product TEXT,
+            address TEXT,
+            extra_data TEXT,
+            FOREIGN KEY(identifier) REFERENCES history_events(identifier) ON UPDATE CASCADE ON DELETE CASCADE
+        );
+    """)  # noqa: E501
+
+    tuples_to_insert = []
+    for (identifier, counterparty, extra_data) in dropped_data:
+        protocol = None
+        address = None
+        if counterparty is not None:
+            if counterparty.startswith('0x'):
+                address = counterparty
+            else:
+                protocol = counterparty
```

Comment:
wouldn't it be better to just re-decode everything? Otherwise you are missing information. I would do this for customized events. There we have to think a bit more

Reply:
Yes, is true that we lose information in the current implementation.

My thought here was that with redecoding we will only populate `address` field, which is currently needed for decoding purposes and doesn't make any change to the UI / accounting / etc.

Reply:
Okay